presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -94,6 +95,10 @@
      */
     TableHandle getAlternativeTableHandle(Session session, TableHandle tableHandle, PartitioningHandle partitioningHandle);
 
+    boolean isPushdownFilterSupported(Session session, TableHandle tableHandle);
```

Comment:
When filter node and scan node get moved to SPI, this interface can be replaced with a connector optimizer rule (by moving the predicate in filter node to table scan's table layout). So shall we add a javadoc to these two interfaces to indicate the interfaces will be removed once we have plan node in SPI (with a ticket number 12546) maybe?

Reply:
ditto for `pushdownFilter`

Reply:
We can add it now if we merge #12904 :D